Cargo.toml

```
@@ -48,10 +48,11 @@ tokio = { version = "0.2", features = ["full"] }
 async-std = { version = "1.5.0", features = ["attributes"] }
 
 [features]
-default = ["http", "ws", "http-tls", "ws-tls"]
-http = ["hyper", "url", "base64"]
+default = ["http", "ws", "http-tls", "ws-tls", "test"]
+http = ["hyper", "url", "base64", "test"]
```

```suggestion
default = ["http", "ws", "http-tls", "ws-tls"]
http = ["hyper", "url", "base64"]
```

Let's disable `test` by default, but change `cfg` stuff to enable it for tests as well to make sure they work without enabling any extra features.

---

I'm wondering why we wouldn't want this by default since we nearly always want unit tests. Is there another way to write unit tests that I haven't found?

---

Or that what you meant, by default, the feature is enabled but only for tests?

---

Yes, that's why I suggested the `#[cfg(any` stuff. Even though it might be useful for external developers to have the test transport, they can enable it only for `dev-dependencies`. AFAIK current cargo does not conflate the features between these deps any more, so it would be cleaner.